obsolete/weather/weather/wm918/wx200d-1.2/wxdebug.1
.\" @(#)$Id: wxdebug.1,v 1.1 2002/09/23 19:12:51 bogdan Rel $
.\"
.\" Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>
.\"
.\" (see the files README and COPYING for more details)

.TH WXDEBUG 1 "Feb 01, 2000" "wx200d Manual"
.SH NAME
wxdebug \- show contents of WX200 data groups
.SH SYNOPSIS
.B wxdebug
[-h] [-01px89abcfu]

.SH DESCRIPTION

.B wxdebug
takes binary WX200 data on the standard input and produces human
readable data group listings on the standard output.  Groups are
listed one per line.  Any or all groups may be shown.
.B wxdebug
can be used with
.B wxstdout(1)
to monitor real-time WX200 data, or with
.B cat(1)
or
.B zcat(1)
to look at binary
.B wx200d(8)
data files.  See
.I EXAMPLES
below.
.P

The data group always appears first on the line.  The hexidecimal
display mode displays each byte of the group in hexidecimal.  The
plain text display mode shows all known data after parsing as
integers, but before any decimal point correction or unit conversions.
For example temperature is shown in tenths of degrees Celsius while
temperature alarms are shown in degrees Fahrenheit, since this is the
way the WX200 has sent them.
.P

The unknowns are displayed as (byte #) = (hex value) and the groups of
unknowns are separate by a / character.
.P

Groups 8, 9, and A show indoor data first, followed by outdoor data.
Timestamps are always in the form: MM/DD:HH:MM:SS.  The general layout
of related data is:
.P

(alarm range) [=] (low) (low timestamp) < (current value) < (high)
(high timestamp)
.P

The following characters have the given meanings (in certain groups).

.TP 0.5i
.B a
The following number(s) is(are) the alarm setting and the alarm is OFF.

.TP 0.5i
.B A
The following number(s) is(are) the alarm setting and the alarm is ON.

.TP 0.5i
.B f
The following number is the format (unit) setting.

.TP 0.5i
.B e
The following sensor value is not in error.

.TP 0.5i
.B E
The following sensor value is in error.

.TP 0.5i
.B o
The preceeding value is in range.

.TP 0.5i
.B O
The preceeding value is out of range.

.TP 0.5i
.B l
The following numer is the local pressure (group A).

.TP 0.5i
.B s
The following numer is the sea level pressure (group A).  The
following numbers are the screen being displayed (group C).

.TP 0.5i
.B p
The following numer is the prediction (group A).  The following number
is the power indicator (group C).

.TP 0.5i
.B b
The following number is the battery indicator (group C).

.TP 0.5i
.B t
The following numer is the pressure trend (group A).  The following
number is the total rain (group B).

.TP 0.5i
.B r
The following number is the rain rate (group B).

.TP 0.5i
.B y
The following number is yesterday's rain amount (group B).

.TP 0.5i
.B g
The following numbers are the gust wind direction @ speed ~ average
wind direction @ speed (group C).

.SH OPTIONS

.TP 0.5i
.B -h
Display a help usage message showing these options, then exit.

.TP 0.5i
.B -0
Turn all groups off.

.TP 0.5i
.B -1
Turn all groups on.  This is the default.

.TP 0.5i
.B -p
Toggle plain text group display mode.  This is on by default.

.TP 0.5i
.B -x
Toggle hexidecimal group display mode.  This is off by default.

.TP 0.5i
.B -8
Toggle display of Time/Humidity group 8.

.TP 0.5i
.B -9
Toggle display of Temperature group 9.

.TP 0.5i
.B -a
Toggle display of Barometer/Dew Point group A.

.TP 0.5i
.B -b
Toggle display of Rain group B.

.TP 0.5i
.B -c
Toggle display of Wind/Wind Chill group C.

.TP 0.5i
.B -f
Toggle display of Timestamp group F.

.TP 0.5i
.B -u
Toggle display of bytes that may contain unknown bits.  All bits of
the bytes are shown, whether known or not.  See wx200.txt for details
of which bits are still unknown.  Please let me know if you discover
what an unknown bit is!

.SH EXAMPLES

.TP 0.5i
.B wxstdout | wxdebug
Monitor all groups of the current WX200 data in real-time.

.TP 0.5i
.B wxstdout | wxdebug -1cf
Monitor all groups except wind and timestamps of the current WX200
data in real-time.

.TP 0.5i
.B zcat 19980206.bin.gz | wxdebug -x0c
Show wind group of given binary data file in both hexidecimal and
parsed formats.

.SH LIMITATIONS

.SH BUGS
Possibly.

.SH SEE ALSO
wxstdout(1), wxfilter(1), wx200d(5), wx200d(8)

.SH AUTHOR
Tim Witham <twitham@quiknet.com>
.SH COPYRIGHT
Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>

wxdebug is released under the conditions of the GNU General Public
License.  See the files README and COPYING in the distribution for
details.


sip_router/NEWS
XXX content of this file reflects a future release;
    features described here may or may not be available
    in the CVS tree
    

Release notes for SIP Express Router (ser)
***********************************************

$Id: NEWS,v 1.12 2003/04/18 14:11:30 jiri Exp $

***********************************************
* Changes introduced in 0.8.11
***********************************************

+--------------------------------------------------------+
| CAUTION: the 0.8.11 release include changes which      |
| are incompatible with scripts and databases used       |
| in previous versions. Care is advised when upgrading   |
| from previous releases to 0.8.11.                      |
+--------------------------------------------------------+

New features
=============
- RFC3261 support
	- TCP support and cross-transport forwarding [core]
	- loose routing support [rr module]
- New modules
	- vm -- voicemail interface [vm]
	- ENUM support [enum]
	- presence agent [pa]
	- dynamic domain management -- allows to manipulate 
  	  hosting of multiple domains in run-time [module]
	- flat-text-file database support [dbtext]
	- rich access control lists [permissions]
- Feature Improvements
	- click-to-dial, which is based on improved tm/FIFO 
	  that better supports external applications [tm module]
	- web accounting -- acc module can report to serweb
  	  on placed calls [acc module]
	- improved exec module (header fields passed now
      as environment variables to scripts) [exec module]
- Architectural Improvements
	- powerpc fast locking support
	- netbsd support
	- 64 bits arch. support (e.g. netbsd/sparc64).
- New Experimental Features (not tested at all yet)
	- nathelper utility for Cisco/ATA NAT traversal [nathelper]
	- another NAT traversal utility [mangler]
	- postgress support [postgress]
	- fcp module [fcp]
	- pdt module (prefix2domain) [pdt]

Changes to use of ser scripts
=============================

About Multiple Transport Support
--------------------------------
SER now suports multiple transport protocols: UDP and TCP. As there
may be UAs which support only either protocol and cannot speak to
each other directly, we recommend to alway record-route SIP requests,
to keep the transport-translating SER in path. Also, if a destination
transport is not known, stateful forwarding is recommended -- use of
stateless forwarding for TCP2UDP would result in loss of reliability.


core
----
- reply_route has been renamed to failure_route -- the old name caused
  too much confusion
- forward_tcp and forward_udp can force SER to forward via specific
  transport protocol

acc module:
-----------
- radius and sql support integrated in this module; you need to
  recompile to enable it
- acc_flag is now called log_flag to better reflect it relates
  to the syslog mode (as opposed to sql/radius); for the same
  reasons, the accounting action is now called "acc_log_request"
- log_fmt allows now to specify what will be printed to syslog

auth module:
------------
- auth module has been split in auth, auth_db, auth_radius, group
  group_radius, uri and uri_radius 
- all the parameters that were part of former auth module are now 
  part of auth_db module
XXX 

im module:
----------
- im is no longer used and has been obsoleted by TM

exec module:
------------
- exec_uri and exec_user have been obsoleted by exec_dset; 
  exec_dset is identical to exec_uri in capabilities; it 
  additionaly passes content of request elements (header 
  fields and URI parts) in environment variables; users of 
  exec_user can use exec_dset now and use the "URI_USER"  
  variable to learn user part of URI
- exec_dset and exec_msg return false, if return value of 
  script does not euqal zero
- exec_dset takes an additional parameter, which enables 
  validation of SIP URIs returned by external application

jabber module:
--------------
- presence support for Jabber users is enabled loading the PA
  module and using subscribe("jabber") for SUBSCRIBE requests to
  jabber user 

msilo module:
-------------
- m_store has now a parameter to set what should be considered
  for storing as destination uri. This enables support for saving
  the messages on negative replies.

radius_acc module:
------------------
- radius_acc module has been removed and radius accounting 
  is now part of acc module

registrar/usrloc modules:
-------------------------
XXX
  
rr module:
----------
- addRecordRoute has been replaced with record_route
- rewriteFromRoute has been replaced with loose_route()
- a new option, "enable_full_lr" can be set to make life
  with misimplemented UAs easier and put LR in from "lr=on"

tm module:
----------
- t_reply_unsafe, used in former versions within reply_routes,
  is deprecated; now t_reply is used from any places in script
- t_on_negative is renamed to t_on_failure -- the old name just
  caused too much confusion
- FIFO t_uac used by some applications (like serweb) has been
  replaced with t_uac_dlg (which allows easier use by dialog-
  oriented applications, like click-to-dial) 
- if you wish to do forward to another destination from 
  failure_route (reply_route formerly), you need to call t_relay
  or t_relay_to explicitely now
- t_relay_to has been replaced with t_relay_to_udp and t_relay_to_tcp



obsolete/weather/weather/wm918/wx200d-1.2/wxdebug.c
/*
 * @(#)$Id: wxdebug.c,v 1.1 2002/09/23 19:12:51 bogdan Rel $
 *
 * Copyright (C) 1998 - 2000 Tim Witham <twitham@quiknet.com>
 *
 * (see the files README and COPYING for more details)
 *
 * This file implements a debuging filter (stdin to stdout).
 *
 */

#include <stdio.h>
#include <unistd.h>
#include <getopt.h>
#include "wx200.h"

#define ALARM(flag)	(flag ? 'A' : 'a')
#define OR(flag)	(flag ? 'O' : 'o')
#define ERR(flag)	(flag ? 'E' : 'e')

void
showhex(unsigned char *buf, int num)
{
  int i;
  for (i = 0; i < num; i++) {
    printf("%s%02x", (i % 2 ? " " : ""), buf[i]);
  }
  printf("\n");
}

int
main(int argc, char **argv)
{
  const char *flags = "Hh0189abcfupx";
  int opt, group, eight, nine, a, b, c, f, u, p = 1, x = -1;
  WX_UNK prev;

  eight = nine = a = b = c = f = u = 1;
  while ((opt = getopt(argc, argv, flags)) != EOF) {
    switch (opt) {
    case '?':
    case 'h':
    case 'H':
      printf("usage: debug [-0189abcfupx]\n"
	     "\t0\tall groups off\n\t1\tall groups on (default)\n"
	     "\tpx\ttoggle plain mode (default on) and hex mode (default off)\n"
	     "\t89abcfu\ttoggle group on/off, f=timestamp, u=unknowns\n");
      exit(0);
    case '0':
      eight = nine = a = b = c = f = u = -1;
      break;
    case '1':
      eight = nine = a = b = c = f = u = 1;
      break;
    case '8':
      eight *= -1;
      break;
    case '9':
      nine *= -1;
      break;
    case 'a':
      a *= -1;
      break;
    case 'b':
      b *= -1;
      break;
    case 'c':
      c *= -1;
      break;
    case 'f':
      f *= -1;
      break;
    case 'u':
      u *= -1;
      break;
    case 'p':
      p *= -1;
      break;
    case 'x':
      x *= -1;
      break;
    }
  }
  wx200bufinit();
  while ((group = wx200bufread(0)) != EOF) {
    if (group >= 0)
      wx200parse(group);
    switch (group) {
    case GROUPF:
      if (f < 0) break;
      if (x > 0) showhex(groupbuf[group], grouplen[group]);
      if (p > 0)
	printf("f %02d:%02d:%02d\n",
	       wx.td.clock.hour, wx.td.clock.min, wx.td.clock.sec);
      break;
    case GROUP8:
      if (eight < 0) break;
      if (x > 0) showhex(groupbuf[group], grouplen[group]);
      if (p > 0)
	printf("8 %02d/%02d:%02d:%02d:%02d %c%02d:%02d"
	       " %c%d-%d=%d%c%02d/%02d:%02d:%02d<%c%d%c<%d%c%02d/%02d:%02d:%02d"
	       " %c%d-%d=%d%c%02d/%02d:%02d:%02d<%c%d%c<%d%c%02d/%02d:%02d:%02d"
	       " f%d\n",

	       wx.td.clock.mon, wx.td.clock.day,
	       wx.td.clock.hour, wx.td.clock.min, wx.td.clock.sec,
	       ALARM(wx.td.a.on), wx.td.a.hi, wx.td.a.lo,

	       ALARM(wx.humid.ina.on), wx.humid.ina.lo, wx.humid.ina.hi,
	       wx.humid.inlo.val, OR(wx.humid.inlo.or),
	       wx.humid.inlo.mon, wx.humid.inlo.day,
	       wx.humid.inlo.hour, wx.humid.inlo.min,
	       ERR(wx.humid.in.err), wx.humid.in.val, OR(wx.humid.in.or),
	       wx.humid.inhi.val, OR(wx.humid.inhi.or),
	       wx.humid.inhi.mon, wx.humid.inhi.day,
	       wx.humid.inhi.hour, wx.humid.inhi.min,

	       ALARM(wx.humid.outa.on), wx.humid.outa.lo, wx.humid.outa.hi,
	       wx.humid.outlo.val, OR(wx.humid.outlo.or),
	       wx.humid.outlo.mon, wx.humid.outlo.day,
	       wx.humid.outlo.hour, wx.humid.outlo.min,
	       ERR(wx.humid.out.err), wx.humid.out.val, OR(wx.humid.out.or),
	       wx.humid.outhi.val, OR(wx.humid.outhi.or),
	       wx.humid.outhi.mon, wx.humid.outhi.day,
	       wx.humid.outhi.hour, wx.humid.outhi.min,

	       wx.td.format);
      break;
    case GROUP9:
      if (nine < 0) break;
      if (x > 0) showhex(groupbuf[group], grouplen[group]);
      if (p > 0)
	printf("9"
	       " %c%d-%d=%d%c%02d/%02d:%02d:%02d<%c%d%c<%d%c%02d/%02d:%02d:%02d"
	       " %c%d-%d=%d%c%02d/%02d:%02d:%02d<%c%d%c<%d%c%02d/%02d:%02d:%02d"
	       " f%d\n",


	       ALARM(wx.temp.ina.on), wx.temp.ina.lo, wx.temp.ina.hi,
	       wx.temp.inlo.val, OR(wx.temp.inlo.or),
	       wx.temp.inlo.mon, wx.temp.inlo.day,
	       wx.temp.inlo.hour, wx.temp.inlo.min,
	       ERR(wx.temp.in.err), wx.temp.in.val, OR(wx.temp.in.or),
	       wx.temp.inhi.val, OR(wx.temp.inhi.or),
	       wx.temp.inhi.mon, wx.temp.inhi.day,
	       wx.temp.inhi.hour, wx.temp.inhi.min,

	       ALARM(wx.temp.outa.on), wx.temp.outa.lo, wx.temp.outa.hi,
	       wx.temp.outlo.val, OR(wx.temp.outlo.or),
	       wx.temp.outlo.mon, wx.temp.outlo.day,
	       wx.temp.outlo.hour, wx.temp.outlo.min,
	       ERR(wx.temp.out.err), wx.temp.out.val, OR(wx.temp.out.or),
	       wx.temp.outhi.val, OR(wx.temp.outhi.or),
	       wx.temp.outhi.mon, wx.temp.outhi.day,
	       wx.temp.outhi.hour, wx.temp.outhi.min,

	       wx.temp.format);
      break;
    case GROUPA:
      if (a < 0) break;
      if (x > 0) showhex(groupbuf[group], grouplen[group]);
      if (p > 0)
	printf("a"
	       " %c%d=%d%c%02d/%02d:%02d:%02d<%c%d%c<%d%c%02d/%02d:%02d:%02d"
	       " %c%d=%d%c%02d/%02d:%02d:%02d<%c%d%c<%d%c%02d/%02d:%02d:%02d"
	       " l%c%d%c s%c%d%c p%dt%df%d\n",

	       ALARM(wx.dew.ina.on), wx.dew.ina.lo,
	       wx.dew.inlo.val, OR(wx.dew.inlo.or),
	       wx.dew.inlo.mon, wx.dew.inlo.day,
	       wx.dew.inlo.hour, wx.dew.inlo.min,
	       ERR(wx.dew.in.err), wx.dew.in.val, OR(wx.dew.in.or),
	       wx.dew.inhi.val, OR(wx.dew.inhi.or),
	       wx.dew.inhi.mon, wx.dew.inhi.day,
	       wx.dew.inhi.hour, wx.dew.inhi.min,

	       ALARM(wx.dew.outa.on), wx.dew.outa.lo,
	       wx.dew.outlo.val, OR(wx.dew.outlo.or),
	       wx.dew.outlo.mon, wx.dew.outlo.day,
	       wx.dew.outlo.hour, wx.dew.outlo.min,
	       ERR(wx.dew.out.err), wx.dew.out.val, OR(wx.dew.out.or),
	       wx.dew.outhi.val, OR(wx.dew.outhi.or),
	       wx.dew.outhi.mon, wx.dew.outhi.day,
	       wx.dew.outhi.hour, wx.dew.outhi.min,

	       ERR(wx.baro.local.err), wx.baro.local.val, OR(wx.baro.local.or),
	       ERR(wx.baro.sea.err), wx.baro.sea.val, OR(wx.baro.sea.or),

	       wx.baro.pred, wx.baro.trend, wx.baro.format);
      break;
    case GROUPB:
      if (b < 0) break;
      if (x > 0) showhex(groupbuf[group], grouplen[group]);
      if (p > 0)
	printf("b %c%d r%c%d%c y%d%c t%d%c %02d/%02d:%02d:%02d f%d\n",
	       ALARM(wx.rain.a.on), wx.rain.a.hi,
	       ERR(wx.rain.rate.err), wx.rain.rate.val, OR(wx.rain.rate.or),
	       wx.rain.yest.val, OR(wx.rain.yest.or),
	       wx.rain.total.val, OR(wx.rain.total.or),
	       wx.rain.total.mon, wx.rain.total.day,
	       wx.rain.total.hour, wx.rain.total.min, wx.rain.format);
      break;
    case GROUPC:
      if (c < 0 && u < 0) break;
      if (c > 0 && x > 0) showhex(groupbuf[group], grouplen[group]);
      if (c > 0 && p > 0)
	printf("c %c%d"
	       " g%c%03d@%c%d%c~%c%03d@%c%d%c<%03d@%d%c=%02d/%02d:%02d:%02d"
	       " %c%d %d%c%02d/%02d:%02d:%02d<%c%d%c"
	       " p%d b%d s%d/%d/%d f%d\n",

	       ALARM(wx.wind.a.on), wx.wind.a.hi,

	       ERR(wx.wind.gdir.err), wx.wind.gdir.val,
	       ERR(wx.wind.gspeed.err), wx.wind.gspeed.val,
	       OR(wx.wind.gspeed.or),
	       ERR(wx.wind.adir.err), wx.wind.adir.val,
	       ERR(wx.wind.aspeed.err), wx.wind.aspeed.val,
	       OR(wx.wind.aspeed.or),
	       wx.wind.dirhi.val, wx.wind.speedhi.val,
	       OR(wx.wind.speedhi.or),
	       wx.wind.speedhi.mon, wx.wind.speedhi.day,
	       wx.wind.speedhi.hour, wx.wind.speedhi.min,

	       ALARM(wx.chill.a.on), wx.chill.a.lo,
	       wx.chill.low.val, OR(wx.chill.low.or),
	       wx.chill.low.mon, wx.chill.low.day,
	       wx.chill.low.hour, wx.chill.low.min,
	       ERR(wx.chill.temp.err), wx.chill.temp.val, OR(wx.chill.temp.or),

	       wx.gen.power, wx.gen.lowbat,
	       wx.gen.section, wx.gen.screen, wx.gen.subscreen, wx.wind.format);
      if (u > 0 && p > 0) {
	if (memcmp(&wx.unk, &prev, sizeof(wx.unk))) {
	  printf("u %02d/%02d:%02d:%02d:%02d "
		 "5=%x,32=%x,33=%x/13=%x,15=%x,28=%x,30=%x,31=%x"
		 "/5=%x,6=%x,28=%x,29=%x/2=%x,10=%x,12=%x"
		 "/14=%x,15=%x,21=%x,23=%x,24=%x,25=%x\n",
		 wx.td.clock.mon, wx.td.clock.day,
		 wx.td.clock.hour, wx.td.clock.min, wx.td.clock.sec,
		 wx.unk.eightf5, wx.unk.eightf32, wx.unk.eightf33,
		 wx.unk.ninef13, wx.unk.ninef15, wx.unk.ninef28,
		 wx.unk.ninef30, wx.unk.ninef31, wx.unk.af5, wx.unk.af6,
		 wx.unk.af28, wx.unk.af29, wx.unk.bf2, wx.unk.bf10,
		 wx.unk.bf12, wx.unk.cf14, wx.unk.cf15, wx.unk.cf21,
		 wx.unk.cf23, wx.unk.cf24,wx.unk.cf25);
	  memcpy(&prev, &wx.unk, sizeof(wx.unk));
	}
      }
      break;
    }
  }
  exit(0);
}
